riscv/mmu.h

```diff
@@ -406,6 +406,7 @@ class mmu_t
   inline bool in_mprv()
   {
     return proc != nullptr
+           && !(proc->state.mnstatus && !get_field(proc->state.mnstatus->read(), MNSTATUS_NMIE))
```

Comment:
Would it make sense/improve anything if we always instantiated `state.mnstatus` and then only put it into `csrmap` if Smrnmi is enabled. Then we wouldn't need to test for null in at least three places I see.

Reply:
The only reason I didn't do that to begin with is that the reset value of the NMIE bit is 0 if the extension is implemented, but to suppress the behavior we need to set it to 1.  I'll try it and see how I feel about it.

Reply:
I don't think it turns out to be an improvement.  Since we need NMIE to be 1 to suppress the new behavior, we'll have a conundrum if we ever add devices to Spike that can generate NMIs: NMIs would appear to be enabled even though the extension isn't present, so we'd need a special-case check over there instead.